### PR TITLE
Add ContentType and extra field types to Attribute

### PR DESCRIPTION
**Pull request by mtahirtariq**
Adds:
- ContentType ForeignKey in Attribute Model
- Update Entity class to pass entity to get_all_attributes method.
- Add new Attribute field types
  - decimal
  - enum_multi

## Motivation and Context
Attributes are shared across all the models which register with them. This PR adds support to assign specific Attributes to specific models.

## Types of Changes
* What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
 Go over all the following points, and put an `x` in all the boxes that apply.
If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: eav/models.py [hunk 1]
```diff
@@ -59,7 +61,8 @@ class EnumValue(models.Model):
        only have a total of four *EnumValues* objects, as you should have used
        the same *Yes* and *No* *EnumValues* for both *EnumGroups*.
     """
-    value = models.CharField(_('Value'), db_index=True, unique=True, max_length=50)
+    value = models.CharField(_('Value'), db_index=True, max_length=100)
+    legacy_value = models.CharField(_('Legacy Value'), blank=True, null=True, db_index=True, max_length=100)
```
Comment (ragdubizzle) [inline eav/models.py:65]: What is the purpose of this field?

File: eav/models.py [hunk 2]
```diff
@@ -131,30 +134,49 @@ class Attribute(models.Model):
...
-    TYPE_ENUM    = 'enum'
+    TYPE_TEXT       = 'text'
+    TYPE_FLOAT      = 'float'
+    TYPE_DECIMAL    = 'decimal'
+    TYPE_INT        = 'int'
+    TYPE_DATE       = 'date'
+    TYPE_BOOLEAN    = 'bool'
+    TYPE_OBJECT     = 'object'
+    TYPE_ENUM       = 'enum'
+    TYPE_ENUM_MULTI = 'enum_multi'
```
Comment (ragdubizzle) [inline eav/models.py:148]: This kind of aligning is against pep8, no?

File: eav/models.py [hunk 3]
```diff
@@ -131,30 +134,49 @@ class Attribute(models.Model):
...
 
     # Core attributes
+    entity_ct = models.ForeignKey(
+        ContentType,
+        on_delete    = models.PROTECT,
+        related_name = 'attribute_entities',
+        blank=True,
+        null=True,
+    )
+    entity_id = models.UUIDField(
```
Comment (ragdubizzle) [inline eav/models.py:170]: Why is this a UUID field?

File: eav/models.py [hunk 4]
```diff
@@ -131,30 +134,49 @@ class Attribute(models.Model):
...
+        (TYPE_BOOLEAN,    _('True / False')),
+        (TYPE_OBJECT,     _('Django Object')),
+        (TYPE_ENUM,       _('Choice')),
+        (TYPE_ENUM_MULTI, _('Multiple Choice')),
     )
 
     # Core attributes
+    entity_ct = models.ForeignKey(
+        ContentType,
+        on_delete    = models.PROTECT,
```
Comment (ragdubizzle) [inline eav/models.py:165]: too many spaces before =

File: eav/models.py [hunk 5]
```diff
@@ -131,30 +134,49 @@ class Attribute(models.Model):
...
+        blank=True,
+        null=True,
+    )
+    entity_id = models.UUIDField(
+        blank=True,
+        null=True,
+    )
+    entity = generic.GenericForeignKey(
+        ct_field = 'entity_ct',
+        fk_field = 'entity_id'
```
Comment (ragdubizzle) [inline eav/models.py:176]: nit: space around `=`

File: eav/models.py (hunk 1 above)
Comment (mtahirtariq) [reply eav/models.py:65]: It will be used to synchronize existing data.

**Pull request merged**